src/python/pants/backend/javascript/nodejs_project.py

```diff
@@ -89,7 +89,7 @@ def extra_env(self) -> dict[str, str]:
 
     def extra_caches(self) -> dict[str, str]:
         if self.package_manager == "pnpm":
-            return {"pnpm_home": "{chroot}/._pnpm_home"}
+            return {"pnpm_home": "._pnpm_home"}
```

Comment:
How big of a lift would it be to add a regression test for this? I see that `{chroot}` is still used in `extra_env` above to point at this link dir and it's not clear to me if that should also be changed or if everything is now working as intended - a test would help me be sure it's good without needing to trace through everywhere `{chroot}` gets substituted.

Reply:
Speculation: A test testing that append only caches is populated will be flaky / cause false negatives, because the cache may be populated by a different process or by a previous run of the test case. IIUIC this cache is persistent and per repository.

It might still be worth to write such a test, but it isnt obviously worthwile to me.

Reply:
Maybe the location of the append only cache dir can be a test-specific directory, now that I think about it..

Reply:
As for the question whether `{chroot}` should be part of the env, the awnser is yet. The `append_only_caches` are chroot relative, env vars are not.

Reply:
@danxmoran https://github.com/pantsbuild/pants/pull/18921/commits/947ff67439bc4016d4f580c3dcb9652be598d628 thoughs?